ComponentKit/Core/CKDimension.mm

```diff
@@ -131,12 +131,11 @@ _Range intersect(const _Range &other) const
 }
 size_t CKSizeRange::hash() const
 {
-  std::hash<CGFloat> hasher;
   NSUInteger subhashes[] = {
-    hasher(min.width),
-    hasher(min.height),
-    hasher(max.width),
-    hasher(max.height)
+      NSUInteger(min.width),
+      NSUInteger(min.height),
+      NSUInteger(max.width),
+      NSUInteger(max.height)
```

Comment:
I just reviewed the original way and I think the issue could probably be that we throw a subhashes with huge values as elements and do four times of 8 bits shift left in CKIntegerArrayHash.

For example, let's say there is a width value which is 375 here. Next have it via hasher and get a new value 1136361472.

```
std::hash<float> hasher;
hasher(CGFloat(375)); // <--- The value is 1136361472
```

If the value is the first element of subhashes, let's imagine how big the final result will be after doing four times of bit shift left with 8 bits, especially comparing to NSUIntegerMax which is 4294967295.

Therefore, to cast geometry from CGFloat into NSUInteger as elements in subhashes can avoid to dealing with a value which is huge already; however, one thing has to be noted that the count of subhashes has to be under a certain limit. Ex. There are 10 elements in subhashes. The same problem will occur again even values are small, like 375 etc. 

For now, I guess current CKIntegerArrayHash can handle well because I didn’t find other subhashes with a huge value as the first element in componentKit. However, it could be better to take the variable "count" of CKIntegerArrayHash into account since it’s in function already.

Have a nice weekend!!

Allen